exporter/exporterhelper/internal/persistent_queue.go

```diff
@@ -41,6 +42,8 @@ import (
 //	 index          index   x
 //	                        xxxx deleted
 type persistentQueue[T any] struct {
+	QueueCapacityLimiter[T]
```

Comment:
Why embedding ?

Reply:
To have the `Capacity()` implemented by the limiter